persistent/Database/Persist/Class/PersistQuery.hs

```diff
@@ -54,6 +57,31 @@ class (PersistCore backend, PersistStoreRead backend) => PersistQueryRead backen
     exists :: (MonadIO m, PersistRecordBackend record backend)
            => [Filter record] -> ReaderT backend m Bool
 
+-- | Backends supporting conditional read operations, which may also support
+-- streaming the results in a memory-constant way.
+--
+-- @since 2.13.2.0
+class (PersistQueryRead backend) => PersistQueryStream backend where
```

Comment:
I think adding this class makes the expectations around `selectSource` *really* weird - that's a class method on `PersistQuery`, so why do we have a separate method here?